libs/x-content/src/main/java/org/opensearch/common/xcontent/smile/SmileXContent.java

```diff
@@ -79,6 +79,7 @@ public static XContentBuilder contentBuilder() throws IOException {
         smileFactory.configure(JsonParser.Feature.STRICT_DUPLICATE_DETECTION, true);
         smileFactory.setStreamReadConstraints(StreamReadConstraints.builder().maxStringLength(DEFAULT_MAX_STRING_LEN).build());
         smileFactory.configure(StreamReadFeature.USE_FAST_DOUBLE_PARSER.mappedFeature(), true);
+        smileFactory.configure(SmileGenerator.Feature.LENIENT_UTF_ENCODING, true);
```

Comment:
Why only for `Smile`? It seems like CBOR has to be tweaked as well?

Reply:
Thanks @reta for your comment, good point, yes I will modify CBOR as well.